MekHQ/src/mekhq/campaign/force/StrategicFormation.java

```diff
@@ -566,35 +581,37 @@ public static StrategicFormation generateInstanceFromXML(Node wn) {
      * Worker function that calculates the total weight of a force with the given ID
      *
      * @param campaign       Campaign in which the force resides
-     * @param forceId Force for which to calculate weight
+     * @param forceId        Force for which to calculate weight
      * @return Total force weight
      */
     public static double calculateTotalWeight(Campaign campaign, int forceId) {
         double weight = 0.0;
 
         for (UUID id : campaign.getForce(forceId).getUnits()) {
-            Unit unit = campaign.getUnit(id);
-            if (null != unit) {
+            try {
+                Unit unit = campaign.getUnit(id);
```

Comment:
Shorter and neater.

I usually prefer to have this kind of try-catch-continue to use be changed by a `getOptional`, so you let the place that is closes to the error deal with the error (in this case a getUnitOptional inside campaign would throw the error, capture it, log it, and return an optional empty)

But its just preference.